src/shared/ui/components/BaseHash.vue

```diff
@@ -47,20 +49,37 @@ async function copy() {
   }
 }
 
-const content = computed(() => {
+type Content = { t: 'plain', value: string } | { t: 'two-line', first: string, second: string };
+
+const content = computed<Content>(() => {
   switch (props.type) {
-    case 'short':
-      return props.hash.slice(0, 4) + '...' + props.hash.slice(-4);
-    case 'medium':
-      return props.hash.slice(0, 10) + '...' + props.hash.slice(-10);
-    case 'two-line':
-      // eslint-disable-next-line no-case-declarations
+    case 'short': {
+      return {
+        t: 'plain',
+        value: props.hash.slice(0, 4) + '...' + props.hash.slice(-4),
+      };
+    }
+    case 'medium': {
+      return {
+        t: 'plain',
+        value: props.hash.slice(0, 10) + '...' + props.hash.slice(-10),
+      };
+    }
+    case 'two-line': {
       const center = Math.ceil(props.hash.length / 2);
 
-      return [props.hash.slice(0, center), h('br'), props.hash.slice(-center + (props.hash.length % 2))];
-
-    default:
-      return props.hash;
+      return {
+        t: 'two-line',
+        first: props.hash.slice(0, center),
+        second: props.hash.slice(-center + (props.hash.length % 2)),
+      };
+    }
+    default: {
+      return {
+        t: 'plain',
+        value: props.hash,
+      };
+    }
```

Comment:
No, you still need to cover the last case, and then, in `default`, add a type guard that `props.type` is `never`.